Tests/BuiltInRulesTests/NestingRuleTests.swift

```diff
@@ -555,4 +555,48 @@ final class NestingRuleTests: SwiftLintTestCase {
 
         verifyRule(description, ruleConfiguration: ["ignore_typealiases_and_associatedtypes": true])
     }
+
+    func testNestingWithoutCodingKeys() {
+        var nonTriggeringExamples = NestingRule.description.nonTriggeringExamples
+        nonTriggeringExamples.append(contentsOf: [
+            .init("""
+                struct Outer {
+                    struct Inner {
+                        enum CodingKeys: String, CodingKey {
+                            case id
+                        }
+                    }
+                }
+                """
+             ),
+        ])
+
+        var triggeringExamples = NestingRule.description.triggeringExamples
+        triggeringExamples.append(contentsOf: [
+            .init("""
+                struct Outer {
+                    struct Inner {
+                        ↓enum Example: String, CodingKey {
+                            case id
+                        }
+                    }
+                }
+            """),
+            .init("""
+                struct Outer {
+                  enum CodingKeys: String, CodingKey {
+                    case id
+                    ↓struct S {}
+                  }
+                }
+            """),
+        ])
```

Comment:
Let's have another example that triggers on a `CodingKeys` enum when the option is set to `false` (the default).